### ISSUE TITLE
Breaks statements like `self.view = view`

### ISSUE THREAD

**Issue by rggjan** [nicklockwood/SwiftFormat#309]
`self.view = abc`

gets changed to 

`view = abc`

with the default options. This is fine. However, this case:

`self.view = view`

gets changed to 

`view = view`

Which then breaks compilation because of the scoping. This case should probably be detected and ignored...

**Comment by nicklockwood** [comment]
@rggjan SwiftFormat definitely doesn't do this in the general case or it would break basically every codebase. I suspect you've hit upon a specific edge case that it doesn't handle correctly.

Can you provide a larger code sample that I can use to reproduce the problem?

**Comment by rggjan** [comment]
OK, here you go. This works, and swiftformat doesn't change anything:

```
class Foo {
    let view = 2
    func bar() {
        let baz = 3

        #if os(OSX)
            let view = self.view
        #elseif os(iOS)
            let view = self.view
        #endif
    }
}
```

However, once the `let baz` line is commented:

```
class Foo {
    let view = 2
    func bar() {
        // let baz = 3

        #if os(OSX)
            let view = self.view
        #elseif os(iOS)
            let view = self.view
        #endif
    }
}
```

It starts changing the line to `view = view`:

```
class Foo {
    let view = 2
    func bar() {
        // let baz = 3

        #if os(OSX)
            let view = view
        #elseif os(iOS)
            let view = view
        #endif
    }
}
```

**Comment by rggjan** [comment]
Removing the `#if` also works... so it seems to be a combination of having the `#if` and having no statement before that...

**Comment by nicklockwood** [comment]
@rggjan fixed in 0.35.6. Thanks for the bug report!

**Comment by rggjan** [comment]
Awesome, thanks a lot!
